library/date_functions.php

```diff
@@ -1,149 +1,76 @@
 <?php
 
-// ============================================================
-// dateformat
-//
-// return a formated string for date
-// @args:   string (date string), boolean (include day of week)
-//              (it uses $_SESSION['language_choice'] )
-// @return: $date_string (string) - formated string
-// Cristian Navalici lemonsoftware at gmail dot com
-//
-// For Hebrew must be implemented a special calendar functions
-//
-// 10.07.2007 - dateformat accepts now an argument
-// ============================================================
-
-function dateformat($strtime = '', $with_dow = false)
+/**
+ * Format a date string according to the user's language preference.
+ *
+ * Returns a formatted date string based on the user's language choice stored in
+ * $_SESSION['language_choice']. The format varies by language and can optionally
+ * include the day of the week.
+ *
+ * @param string|int $strtime Unix timestamp or date string. If empty, uses current time.
+ * @param bool $with_dow Whether to include the day of the week in the output.
+ * @return string The formatted date string.
+ *
+ * @author Cristian Navalici lemonsoftware at gmail dot com
+ * @note For Hebrew, displays English calendar, NOT Jewish calendar
+ * @note Last modified 10.07.2007 - dateformat accepts now an argument
+ */
+function dateformat(string|int $strtime = '', bool $with_dow = false): string
 {
-
-// without an argument, display current date
+    // without an argument, display current date
     if (!$strtime) {
         $strtime = strtotime('now');
     }
 
-// string date is formed by
-// $dow + date(day) + $nom + date(year) or similar
-
-// name the day of the week for different languages
-    $day = date("w", $strtime); // 0 sunday -> 6 saturday
+    // name the day of the week for different languages
+    $day = (int) date("w", $strtime); // 0 sunday -> 6 saturday
 
-    switch ($day) {
-        case 0:
-            $dow = xl('Sunday');
-            break;
-        case 1:
-            $dow = xl('Monday');
-            break;
-        case 2:
-            $dow = xl('Tuesday');
-            break;
-        case 3:
-            $dow = xl('Wednesday');
-            break;
-        case 4:
-            $dow = xl('Thursday');
-            break;
-        case 5:
-            $dow = xl('Friday');
-            break;
-        case 6:
-            $dow = xl('Saturday');
-            break;
-    }
+    static $days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];
+    $dow = xl($days[$day]);
```

Comment:
hi @kojiromike , Doesn't matter here since weekdays and months are translated in many places. But in other places where things like this may come up (ie. translating a variable), ideal to have the actual string surrounded by the xl function so it gets captured in the perl script that collects constants (in some places in past, I have even put the string surrounded by the xl() function within a comment so it gets captured).